### PR TITLE
Fix: update calls to `runtime.fastrand` to use `cheaprand` for compatibility with Go 1.22

### PR DESCRIPTION
**Pull request by Aoang**
## Description

update calls to `runtime.fastrand` to use `cheaprand` for compatibility with Go 1.22

In Golang 1.22, the `runtime.fastrand` method was renamed to `cheaprand`. This commit updates all occurrences of `runtime.fastrand` in the project to use the new `cheaprand` method to maintain compatibility with the latest Golang version and address performance concerns.

Refer: https://gist.github.com/Aoang/3dc06f127f3f54507b7e06b7b6550c28

#### Related changes (conditional)

- Tests
    - [x] If I added new functionality, I added tests covering it.
    - [x] If I fixed a bug, I added a regression test to prevent the bug from
      silently reappearing again.

- Documentation
    - [x] I checked whether I should update the docs and did so if necessary:
        - [README](../README.md)

- Public contracts
    - [x] My changes doesn't break project license.

#### Stylistic guide (mandatory)

- [x] My code complies with the [styles guide](https://github.com/uber-go/guide/blob/master/style.md).
- [x] My commit history is clean (only contains changes relating to my
  issue/pull request and no reverted-my-earlier-commit changes) and commit
  messages start with identifiers of related issues in square brackets.

  **Example:** `[#42] Short commit description`

  If necessary both of these can be achieved even after the commits have been
  made/pushed using [rebase and squash](https://git-scm.com/docs/git-rebase).

#### Before merging (mandatory)
- [x] Check __target__  branch of PR is set correctly

### CODE REVIEW

**Comment by maypok86** [comment]
@Aoang 
Oh, it looks really good. Only as far as I know, with Go1.21 the standard `rand.Uint32` started using `fastrand` and `atomic.Load`, which is why it has become ultra fast. Therefore, it may be worth just using the standard function from certain versions of Go.

There also seems to be another incompatibility with Go1.22 due to `swiss.Map` that also uses `fastrand`.

So I would be very grateful if you could replace it [here](https://github.com/maypok86/otter/blob/main/internal/s3fifo/ghost.go#L27) with a standard one. According to my benchmarks, this does not affect overall performance, and in the future we may switch to bloom filters.

File: internal/xruntime/runtime_1.22.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xruntime
+
+import (
```
Comment (maypok86) [inline internal/xruntime/runtime_1.22.go:19]:
```suggestion
import (
    "rand"
    "time"
)

var (
    r = rand.New(rand.NewSource(time.Now().Unix()))
)
```

Comment (Aoang) [reply internal/xruntime/runtime_1.22.go:19]: Why haven't considered using `math/rand/v2`?

Comment (maypok86) [reply internal/xruntime/runtime_1.22.go:19]: Yes, it would probably be better this way, then the global variable is not needed.

**Review by maypok86 — APPROVED** [review]
It looks good to me. You're amazing!

**Pull request merged**